src/transforms/deployment.ts

```diff
@@ -1,7 +1,84 @@
 import _ from "lodash";
-import { Context } from "probot/lib/context";
 import issueKeyParser from "jira-issue-key-parser";
-import { JiraDeploymentData } from "../interfaces/jira";
+import {JiraDeploymentData} from "../interfaces/jira";
+import {GitHubAPI} from "probot";
+import {WebhookPayloadDeploymentStatus} from "@octokit/webhooks";
+import {LoggerWithTarget} from "probot/lib/wrap-logger";
+import {Octokit} from "@octokit/rest";
+import {booleanFlag, BooleanFlags} from "../config/feature-flags";
+
+// https://docs.github.com/en/rest/reference/repos#list-deployments
+async function getLastSuccessfulDeployCommitSha(
+	owner: string,
+	repoName: string,
+	github: GitHubAPI,
+	deployments: Octokit.ReposListDeploymentsResponseItem[],
+	logger
+): Promise<string> {
+
+	try {
+		for (const deployment of deployments) {
+			// Get each deployment status for this environment so we can have their statuses' ids
+			const listDeploymentStatusResponse: Octokit.Response<Octokit.ReposListDeploymentStatusesResponse> = await github.repos.listDeploymentStatuses({
+				owner: owner,
+				repo: repoName,
+				deployment_id: deployment.id,
+				per_page: 100 // Default is 30, max we can request is 100
+			});
+
+			// Find the first successful one
+			const lastSuccessful: Octokit.ReposListDeploymentStatusesResponseItem | undefined = listDeploymentStatusResponse.data.find(deployment => deployment.state === "success");
+			if (lastSuccessful !== undefined) {
+				return deployment.sha;
+			}
+		}
+	} catch (e) {
+		logger?.error(`Failed to get deployment statuses.`);
+	}
+
+
+	// If there's no successful deployment on the list of deployments that GitHub returned us (max. 100) then we'll return the last one from the array, even if it's a failed one.
+	return deployments[deployments.length - 1].sha;
+}
+
+async function getCommitMessagesSinceLastSuccessfulDeployment(
+	owner: string,
+	repoName: string,
+	currentDeploySha: string,
+	currentDeployId: number,
+	currentDeployEnv: string,
+	github: GitHubAPI,
+	logger?: LoggerWithTarget
+): Promise<string | undefined> {
+
+	// Grab the last 100 deployments for this repo
+	const deployments: Octokit.Response<Octokit.ReposListDeploymentsResponse> = await github.repos.listDeployments({
+		owner: owner,
+		repo: repoName,
+		environment: currentDeployEnv,
+		per_page: 100 // Default is 30, max we can request is 100
```

Comment:
This should be lower (say, 10), because we're making a call to get deployment status _for each of those deployments_. We don't want to make 100 calls (that count into the rate limit) for each deployment status event that we receive.